packages/react-components/src/components/ChatMessage/ChatMessageComponentWrapper.tsx

```diff
@@ -0,0 +1,127 @@
+  /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+  onFetchInlineAttachment: (attachments: FileMetadata[], messageId: string) => Promise<void>;
+  /* @conditional-compile-remove(image-gallery) */
+  onInlineImageClicked?: (attachmentId: string, messageId: string) => Promise<void>;
+  /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+  inlineAttachments: Record<string, Record<string, string>>;
+  /* @conditional-compile-remove(mention) */
+  mentionOptions?: MentionOptions;
+};
+
+/**
+ * The wrapper component to display different types of chat message.
+ *
+ * @private
+ */
+export const ChatMessageComponentWrapper = (props: ChatMessageComponentWrapperProps): JSX.Element => {
+  const { message, styles, onRenderMessage, key: messageKey } = props;
+
+  /* @conditional-compile-remove(data-loss-prevention) */
+  // Similar logic as switch statement case 'chat', if statement for conditional compile (merge logic to switch case when stabilize)
+  if (message.messageType === 'blocked') {
+    const myChatMessageStyle =
+      message.status === 'failed'
+        ? styles?.failedMyChatMessageContainer ?? styles?.myChatMessageContainer
+        : styles?.myChatMessageContainer;
+    const blockedMessageStyle = styles?.blockedMessageContainer;
+    const messageContainerStyle = message.mine ? myChatMessageStyle : blockedMessageStyle;
+    return (
+      <FluentChatMessageComponentWrapper {...props} message={message} messageContainerStyle={messageContainerStyle} />
+    );
+  }
+
+  switch (message.messageType) {
+    case 'chat': {
+      const myChatMessageStyle =
+        message.status === 'failed'
+          ? styles?.failedMyChatMessageContainer ?? styles?.myChatMessageContainer
+          : styles?.myChatMessageContainer;
+      const chatMessageStyle = styles?.chatMessageContainer;
+      const messageContainerStyle = message.mine ? myChatMessageStyle : chatMessageStyle;
+      return (
+        <FluentChatMessageComponentWrapper {...props} message={message} messageContainerStyle={messageContainerStyle} />
+      );
+    }
+
+    case 'system': {
+      const messageContainerStyle = styles?.systemMessageContainer;
+      const systemMessageComponent =
+        onRenderMessage === undefined ? (
+          <DefaultSystemMessage {...props} />
+        ) : (
+          onRenderMessage({ ...props, messageContainerStyle }, (props) => <DefaultSystemMessage {...props} />)
+        );
+      return (
+        <div key={messageKey} style={{ paddingTop: '0.5rem' }}>
+          {systemMessageComponent}
+        </div>
+      );
+    }
+
+    default: {
+      // We do not handle custom type message by default, users can handle custom type by using onRenderMessage function.
+      const customMessageComponent = onRenderMessage === undefined ? <></> : onRenderMessage({ ...props });
+      return (
+        <div key={messageKey} style={{ paddingTop: '1rem', paddingBottom: '0.25rem' }}>
```

Comment:
nit: the `style` object should probably be memoized here (though is a relatively low level component so not entirely necessary)